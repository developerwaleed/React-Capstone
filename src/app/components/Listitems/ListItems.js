/* eslint-disable jsx-a11y/control-has-associated-label */
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BsSearch } from 'react-icons/bs';
import { fetchProducts, productFilter } from '../redux/products/products';
import ProductComponent from './ProductComponent';
import styles from './ListItems.module.css';

const ProductPage = () => {
  const productData = useSelector((state) => state.allProducts.products);
  const dispatch = useDispatch();

  useEffect(() => {
    if (productData.length === 0) {
      dispatch(fetchProducts());
    }
  }, []);

  const [search, setSearch] = useState('');

  const handleSearch = (e) => {
    e.preventDefault();
    if (search === '') {
      dispatch(fetchProducts());
    } else {
      dispatch(productFilter(search));
    }
  };

  return (
    <>
      <div className={styles.searchBar}>
        <form onSubmit={handleSearch}>
          <input
            type="text"
            className={styles.searchInput}
            value={search}
            placeholder="Search Products"
            onChange={(e) => setSearch(e.target.value)}
          />
        </form>
        <button className={styles.searchBtn} type="submit">
          <BsSearch />
        </button>
      </div>
      <div className={styles.itemsSection}>
        <ProductComponent />
      </div>
    </>
  );
};

export default ProductPage;
