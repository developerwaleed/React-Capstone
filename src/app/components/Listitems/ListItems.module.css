.searchBar {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-left: 8%;
  gap: 0.3rem;
  margin-top: 5%;
}

.searchInput {
  padding: 0.4rem 0.6rem;
  border-radius: 20px;
  outline: none;
  border: none;
  text-align: center;
  font-style: italic;
}

.searchBtn {
  background: none;
  border: none;
  outline: none;
  color: #fff;
  font-size: 1rem;
  padding-top: 0.3rem;
  padding-left: 0.5rem;
}

.itemsSection {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  column-gap: 1rem;
  row-gap: 2rem;
  padding: 7% 4%;
}

@media screen and (min-width: 768px) {
  .searchBar {
    margin-top: 3%;
    margin-left: 0;
  }

  .searchInput {
    padding: 0.7rem 1.5rem;
    font-size: 1rem;
  }

  .searchBtn {
    font-size: 1.4rem;
  }

  .itemsSection {
     grid-template-columns: repeat(3, 1fr);
    place-items: center;
    column-gap: 1rem;
    row-gap: 2rem;
    padding: 7% 4%;
  }
}
