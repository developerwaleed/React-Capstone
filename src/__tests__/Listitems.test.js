import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Listitems from '../app/components/Listitems/Listitems.js';
import store from '../redux/configureStore';

it('renders properly', () => {
  const tree = renderer.create(
    <Router>
      <Provider store={store}>
        <Listitems />
      </Provider>
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
