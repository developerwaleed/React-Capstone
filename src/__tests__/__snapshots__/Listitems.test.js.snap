// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
Array [
  <div
    className="searchBar"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="searchInput"
        onChange={[Function]}
        placeholder="Search Products"
        type="text"
        value=""
      />
    </form>
    <button
      className="searchBtn"
      type="submit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
        />
      </svg>
    </button>
  </div>,
  <div
    className="itemsSection"
  >
    <div
      className="loadingHome"
    >
      <div
        className="Loader"
      />
    </div>
  </div>,
]
`;
